samples/query-features-with-arcade-expression/src/main/java/com/esri/arcgismaps/sample/queryfeatureswitharcadeexpression/components/QueryFeaturesWithArcadeExpressionViewModel.kt

```diff
@@ -0,0 +1,190 @@
+
+    // create and add a map with with portal item
+    val arcGISMap = ArcGISMap(portalItem)
+
+    // create a map view proxy for handling interactions with the map view
+    val mapViewProxy = MapViewProxy()
+
+    // create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+
+            // get the RPD Beats layer from the map's operational layers
+            policeBeatsLayer = arcGISMap.operationalLayers.firstOrNull { layer ->
+                layer.id == "RPD_Reorg_9254"
+            }
+
+            // update query state, map is ready for user interaction
+            _queryStateFlow.value = QueryState(loadState = LoadState.READY_TO_START)
+        }
+
+        // add the marker graphic to the graphics overlay
+        graphicsOverlay.graphics.add(markerGraphic)
+    }
+
+    /**
+     * Handle a tap on the map view from the user
+     */
+    fun handleTap(point: Point, screenCoordinate: ScreenCoordinate) {
+        // update the marker location to where the user tapped on the map
+        markerGraphic.geometry = point
+        viewModelScope.launch {
+            // centre the viewpoint on where the user tapped on the map
+            mapViewProxy.setViewpointCenter(point)
+
+            // evaluate an Arcade expression on the tapped screen coordinate
+            evaluateArcadeExpression(screenCoordinate)
+        }
+    }
+
+    /**
+     * Evaluates an Arcade expression that returns crime in the last 60 days at the tapped
+     * [screenCoordinate] on the [arcGISMap] with the [policeBeatsLayer] and outputs the result
+     * to the [queryStateFlow] property.
+     */
+    private suspend fun evaluateArcadeExpression(screenCoordinate: ScreenCoordinate) {
+        policeBeatsLayer?.let { layer ->
+            // show the loading spinner as the Arcade evaluation can take time to complete
+            _queryStateFlow.value = QueryState(loadState = LoadState.LOADING)
+
+            // identify the layer and its elements based on the position tapped on the mapView and
+            // get the result
+            val result = mapViewProxy.identify(
+                layer = layer,
+                screenCoordinate = screenCoordinate,
+                tolerance = 12.dp,
+                returnPopupsOnly = false
+            )
+
+            // get the result as an IdentifyLayerResult
+            val identifyLayerResult = result.getOrElse { error ->
+                // if the identifyLayer operation failed show an error and return
+                messageDialogVM.showMessageDialog(
+                    "Error identifying layer:",
```

Comment:
Again, this wording doesn't seem quite right. How about "Error from identify operation"?

Reply:
How about "Error performing identify operation"?